classes/security/Validation.php

```diff
@@ -425,28 +429,30 @@ public static function isSiteAdmin()
     /**
      * Check whether a user is allowed to administer another user.
      *
-     * @param int $administeredUserId User ID of user to potentially administer
-     * @param int $administratorUserId User ID of user who wants to do the administrating
-     *
-     * @return bool True IFF the administration operation is permitted
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * @param array $allowedLevels          The allowed adminstration levels
+     * 
+     * @return int The authorized adminstrative level 
      */
-    public static function canAdminister($administeredUserId, $administratorUserId)
+    public static function canAdminister($administeredUserId, $administratorUserId, $allowedLevels = [self::ADMINISTRATION_FULL])
```

Comment:
Hmm, this feels a little bit arbitrary to me. As a coder I would expect `$allowedLevels` to essentially be an optional `array_intersect`, but here it's coded to default to just `ADMINISTRATION_FULL`. I think this is trying to do both parts of https://github.com/pkp/pkp-lib/pull/8283#discussion_r979016275 when it should be either one or the other; the combination is complicating things without preserving backwards compatibility.

Reply:
> this feels a little bit arbitrary to me
> think this is trying to do both suggestions in https://github.com/pkp/pkp-lib/pull/8283#discussion_r979016275 at once, when it should be either one or the other;

@asmecher  I agree 100% with that . So I propose that we introduce a new method `getAdministrationLevel` which will return appropriate level and deprecate the old `canAdminister` method but keep it for now(but marked it as `deprecated`) . that way we can ensure that no breaking change has introduced. Also the previous `canAdminister` was ok as that that time we have only `true/false` but now we have different levels . I have updated it accordingly . please review .

Reply:
@touhidurabir Did that proposal get pushed up? I don't see `getAdministrationLevel` implemented here.

Reply:
@asmecher I can confirm that I pushed those changes yesterday but weirdly they were not present when I checked today . More weird is that even that commit is missing in the commit history . Any idea what may cause this ? BTW, I have again pushed those changes they are there now .